hawkmoth/__init__.py

```diff
@@ -67,8 +67,20 @@ def __parse(self, viewlist, filename):
 
         compat = self.options.get('compat', env.config.cautodoc_compat)
         clang = self.options.get('clang', env.config.cautodoc_clang)
+        transformations = env.config.cautodoc_transformations
 
-        transform = lambda x: doccompat.convert(x, mode=compat)
+        if transformations is not None:
+            # compat == 'none' works by coincidence if there is no 'none'
+            fn = transformations.get(compat, None)
+            if fn is not None:
+                transform = lambda x: fn(x, mode=compat)
```

Comment:
It's weird to derive a function from `compat` above only to then call a different entry point with `mode=compat`. It seems we can get rid oif the `convert()` function and do directly `transform = transformations.get(compat, None)`

Reply:
Actually I was partially off, I was confused by the passing of `compat` into the function. Maybe a remnant of using `convert()`?

Unless we want to pass additional options to the convert function from the user (probably a good and easy thing to do), there's still no need to capture this lambda though.